stream/src/main/scala/org/apache/pekko/stream/impl/UnfoldResourceSource.scala

```diff
@@ -33,66 +34,69 @@ import pekko.stream.stage._
     extends GraphStage[SourceShape[T]] {
   val out = Outlet[T]("UnfoldResourceSource.out")
   override val shape = SourceShape(out)
-  override def initialAttributes: Attributes = DefaultAttributes.unfoldResourceSource
+  override def initialAttributes: Attributes =
+    DefaultAttributes.unfoldResourceSource and SourceLocation.forLambda(create)
 
-  def createLogic(inheritedAttributes: Attributes) = new GraphStageLogic(shape) with OutHandler {
-    lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
-    var open = false
-    var blockingStream: S = _
-    setHandler(out, this)
+  def createLogic(inheritedAttributes: Attributes) =
+    new GraphStageLogic(shape) with OutHandler {
+      private lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
+      private var open = false
+      private var resource: S = _
 
-    override def preStart(): Unit = {
-      blockingStream = create()
-      open = true
-    }
+      override def preStart(): Unit = {
+        resource = create()
+        open = true
+      }
 
-    @tailrec
-    final override def onPull(): Unit = {
-      var resumingMode = false
-      try {
-        readData(blockingStream) match {
-          case Some(data) => push(out, data)
-          case None       => closeStage()
-        }
-      } catch {
-        case NonFatal(ex) =>
-          decider(ex) match {
-            case Supervision.Stop =>
-              open = false
-              close(blockingStream)
-              failStage(ex)
-            case Supervision.Restart =>
-              restartState()
-              resumingMode = true
-            case Supervision.Resume =>
-              resumingMode = true
+      @tailrec
+      final override def onPull(): Unit = {
+        var resumingMode = false
+        try {
+          readData(resource) match {
+            case Some(data) => push(out, data)
+            case None       => closeStage()
           }
+        } catch {
+          case NonFatal(ex) =>
+            decider(ex) match {
+              case Supervision.Stop =>
+                open = false
```

Comment:
indent

Reply:
It's formatted with scala-cli

Reply:
You can review it with `whitespace:off` in the review tab.